### ISSUE TITLE
SingleQuery and grid.get('total')

### ISSUE THREAD

**Issue by AndrewWSHenry** [SitePen/dgrid#1271]
This is more of an issue on a like to have/request.

First, thanks for the SingleQuery [tutorial ](http://dgrid.io/tutorials/1.0/single_query/)at the dgrid.io site. It's a useful mixin for the in-memory store I was working with. Would it be possible to add this mixin to dgrid itself? It's usefulness suggests that it is worthy of inclusion.

And the issue I had was that I needed to display the total number of rows after setting a filtered collection to the grid. I came across the [issue ](https://github.com/SitePen/dgrid/issues/232)that pointed me to using grid.get('total'). After discovering that get('total') always returned zero with the SingleQuery mixin, I looked at the code in _StoreMixin and realized that the _total property would never be set.

I'd like to suggest updating the SingleQuery [tutorial ](http://dgrid.io/tutorials/1.0/single_query/)to include setting total in the renderArray function, so that grid.get('total') functions properly.

```
renderArray: function () {

    var rows = this.inherited(arguments);

	// Clear _lastCollection which is ordinarily only used for
	// store-less grids

	this._lastCollection = null;

	// Set the total number of rows. _StoreMixin has a property called _total.
	// SingleQuery extends _StoreMixin so it can set _total directly.
	// Can then use get('total') that is made available by the _getTotal of _StoreMixin.

	this._total = rows.length;									   

	return rows;
}
```

**Comment by kfranqueiro** [comment]
Thanks for pointing out the omission of setting `_total`.  I've committed updates to the 0.4 and 1.0 versions of the tutorials for that for the next time we update the site.

As for why it's not in dgrid as a first-class mixin, the main reason was to avoid the footgun of people finding it in plain sight, then using it for inappropriately large data sets (or with stores that ordinarily would be used to request a small range at a time from a server), and assuming dgrid is to blame for the ensuing performance nightmare.

However, at this point it sort of _is_ in the repository anyway (tucked under the folder for the laboratory demo which makes use of it), which kind of seems silly.  I'm planning to have another discussion with the team as to whether it's time to just promote it after all (maybe with a warning if it detects a large data set), at least for 1.x.

**Comment by kfranqueiro** [comment]
We've decided to go ahead and add `SingleQuery` as a top-level module in dgrid for the next release, which we're planning to tag as 1.1.0 as a result.  (There are no breaking changes.)  I'll work on moving it and adding tests/docs sometime this week.

**Comment by AndrewWSHenry** [comment]
Cool. Thanks Ken.

**Comment by kfranqueiro** [comment]
`SingleQuery` will be available as `dgrid/extensions/SingleQuery` in 1.1.0:  https://github.com/SitePen/dgrid/commit/fe8092c7639c545dbc86039cec6c2f23363cc3da

Adding this (along with support for things like `noDataMessage` and `loadingMessage`) also enabled us to add some tests and refactor some code, which led to finding some other stuff going on as well, relevant to #1275 and #1277, which will both be fixed in that release.
